updater/lib/dependabot/updater/operations/update_all_versions.rb

```diff
@@ -250,24 +320,28 @@ def log_requirements_for_update(requirements_to_unlock, checker)
 
         # If a version update for a peer dependency is possible we should
         # defer to the PR that will be created for it to avoid duplicate PRs.
+        sig { params(dependency_name: String, updated_deps: T::Array[Dependabot::Dependency]).returns(T::Boolean) }
         def peer_dependency_should_update_instead?(dependency_name, updated_deps)
           updated_deps
             .reject { |dep| dep.name == dependency_name }
             .any? do |dep|
               next true if existing_pull_request([dep])
-              next false if dep.previous_requirements.nil?
+
+              dep_previous_requirements = dep.previous_requirements
+              next false if dep_previous_requirements.nil?
```

Comment:
It seems like some unrelated changes like these and the types for all the methods have gotten into this PR? Try to keep extra changes out since it's already quite large of a change.

Reply:
I'm not sure why we needed to fully type this file, it's better to keep a PR focussed on doing one thing. Often typing a file can cause unrelated problems. Partially typing new methods is preferred in this case.

Reply:
I just replaced it to make it T.must since above we already have nil check.